eng/common/scripts/SemVer.ps1

```diff
@@ -114,6 +134,13 @@ class AzureEngSemanticVersion {
     if ($this.IsPrerelease)
     {
       $versionString += $this.PrereleaseLabelSeparator + $this.PrereleaseLabel + $this.PrereleaseNumberSeparator + $this.PrereleaseNumber
+      if ($this.HasBuildNumber) {
+        if ([AzureEngSemanticVersion]::ParseLanguage -eq "python") {
+          $versionString += $this.BuildNumberSeparator + ("{0:d3}" -f $this.BuildNumber)
```

Comment:
I wonder how much we care about the leading zero spacing here. I'd say lets not. If we do care then we probably also need to care about it for the non python case as well.

Reply:
I just checked and it doesn't look like we currently pad the other languages. For some reason I thought we did. At any case we should be able to handle if there is padding and for the 2 string I'm not sure how much I care about preserving the padding for the python case.

Reply:
I'd rather fit with the spec, as opposed to outputting version strings differently than they come in (we should preserve the input format). The [python version spec](https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#python-1) specifically mentions the zero padding:

> X.Y.ZaYYYYMMDDrrr (rrr is based on the number of builds performed on the given day and it is **zero-padded with a valid range** starting at 001 and ends at 999) used for daily alpha releases.

For other languages, I see no mention of the zero padding (possibly because of the separator?), hence why this is a special case. If you have historical context on this I'd be curious to know :)

Reply:
I wrote the spec (or at least proposed it :)). The reason we did the padding was to ensure these sorted correctly as we couldn't insert a separator.